packages/react-next/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -1,23 +1,30 @@
 import * as React from 'react';
-import { compose } from '@fluentui/react-compose';
+import { compose, mergeProps } from '@fluentui/react-compose';
 import { mergeAriaAttributeValues } from '../../Utilities';
-import { Icon } from '../../Icon';
-import { ICheckboxProps, ICheckboxSlots } from './Checkbox.types';
+import { ICheckboxProps, ICheckboxSlots, ICheckboxState, ICheckboxSlotProps } from './Checkbox.types';
 import { KeytipData } from '../../KeytipData';
 import { useCheckbox } from './useCheckbox';
 
 const defaultSlots: Omit<ICheckboxSlots, 'root'> = {
   input: 'input',
-  // TODO: add checkmark slot in parent component instead
-  checkmark: Icon,
+  checkmark: 'div',
   container: 'label',
   label: 'span',
   checkbox: 'div',
 };
 
 export const CheckboxBase = compose<'div', ICheckboxProps, {}, ICheckboxProps, {}>(
   (props, forwardedRef, composeOptions) => {
-    const { slotProps, slots, state } = useCheckbox(props, composeOptions, forwardedRef);
+    // TODO: improve typing for mergeProps
+    const { slotProps, slots, state } = mergeProps<ICheckboxProps, ICheckboxState>(
+      composeOptions.state,
+      composeOptions,
+    ) as {
+      state: ICheckboxState;
+      slots: ICheckboxSlots;
+      slotProps: ICheckboxSlotProps;
+    };
```

Comment:
Is the `as` needed?

Reply:
yes and i'm addressing this in this PR: #13570 13571